client/core/types.go

```diff
@@ -750,13 +748,6 @@ func (a *dexAccount) ID() account.AccountID {
 	return a.id
 }
 
-// isViewOnly is true if account keys have not been generated AND saved to db.
```

Comment:
I think the reason this PR isn't quite OK is this sorta confusing phrase "AND saved to db."

I'm thinking all these changes would be fine if we made `dbCreateOrUpdateAccount` unnecessary by changing `c.db.UpdateAccountInfo` in to and **upsert** operation, and removed `db.CreateAccount`.  Then things would work fine without the extra flags and such.